service/orchestrator/catalogs.go

```diff
@@ -188,16 +192,27 @@ func (srv *Service) ListControls(_ context.Context, req *orchestrator.ListContro
 
 	// If the category name is set (additional binding), forward it as a condition to the pagination method
 	if req.CategoryName != "" && req.CatalogId != "" {
-		res.Controls, res.NextPageToken, err = service.PaginateStorage[*orchestrator.Control](req, srv.storage,
-			service.DefaultPaginationOpts, "category_name = ? AND category_catalog_id = ?", req.CategoryName, req.CatalogId)
-	} else if req.CatalogId != "" {
-		res.Controls, res.NextPageToken, err = service.PaginateStorage[*orchestrator.Control](req, srv.storage,
-			service.DefaultPaginationOpts, "category_catalog_id = ?", req.CatalogId)
-	} else {
-		res.Controls, res.NextPageToken, err = service.PaginateStorage[*orchestrator.Control](req, srv.storage,
-			service.DefaultPaginationOpts)
+		query = append(query, "category_name = ?")
+		args = append(args, req.CategoryName)
 	}
+	if req.CatalogId != "" {
+		query = append(query, "category_catalog_id = ?")
+		args = append(args, req.CatalogId)
+	}
+
+	// Apply additional filter
+	if req.Filter != nil {
+		if len(req.Filter.AssuranceLevels) > 0 {
+			query = append(query, "(assurance_level IN ? OR assurance_level IS NULL)")
```

Comment:
Why do we need here the part "OR assurance_level is NULL"? I think here it is not possible that the assuranceLevel is null.

Reply:
I think the logic was that if no assurance level is specified (e.g. on the top level), the control is valid for all assurance levels.